src/components/application_manager/src/commands/hmi/navi_set_video_config_request.cc

```diff
@@ -0,0 +1,150 @@
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "application_manager/commands/hmi/navi_set_video_config_request.h"
+#include <string>
+#include <vector>
+
+namespace application_manager {
+
+namespace commands {
+
+NaviSetVideoConfigRequest::NaviSetVideoConfigRequest(
+    const MessageSharedPtr& message, ApplicationManager& application_manager)
+    : RequestToHMI(message, application_manager)
+    , EventObserver(application_manager.event_dispatcher()) {}
+
+NaviSetVideoConfigRequest::~NaviSetVideoConfigRequest() {}
+
+void NaviSetVideoConfigRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (!CheckAvailabilityHMIInterfaces(
+          application_manager_, HmiInterfaces::HMI_INTERFACE_Navigation)) {
+    LOG4CXX_WARN(logger_, "HMI interface Navigation is not supported");
+    return;
+  }
+
+  ApplicationSharedPtr app =
+      application_manager_.application_by_hmi_app(application_id());
+  if (!app) {
+    LOG4CXX_ERROR(logger_,
+                  "Application with hmi_app_id " << application_id()
+                                                 << "does not exist");
+    return;
+  }
+
+  subscribe_on_event(hmi_apis::FunctionID::Navigation_SetVideoConfig,
+                     correlation_id());
+  SendRequest();
+}
+
+void NaviSetVideoConfigRequest::on_event(const event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  ApplicationSharedPtr app =
+      application_manager_.application_by_hmi_app(application_id());
+  if (!app) {
+    LOG4CXX_ERROR(logger_,
+                  "Application is not found, abort NaviSetVideoConfigRequest");
+    return;
+  }
+
+  const smart_objects::SmartObject& message = event.smart_object();
+  switch (event.id()) {
+    case hmi_apis::FunctionID::Navigation_SetVideoConfig: {
+      const hmi_apis::Common_Result::eType code =
+          static_cast<hmi_apis::Common_Result::eType>(
+              message[strings::params][hmi_response::code].asInt());
+      bool result = false;
+      std::vector<std::string> rejected_params;
+
+      if (code == hmi_apis::Common_Result::SUCCESS) {
+        LOG4CXX_DEBUG(logger_, "Received SetVideoConfig success response");
+        result = true;
+      } else {
+        LOG4CXX_DEBUG(logger_,
+                      "Received SetVideoConfig failure response (" << event.id()
+                                                                   << ")");
+        result = false;
+        if (message[strings::msg_params].keyExists("rejectedParams")) {
```

Comment:
Done in https://github.com/smartdevicelink/sdl_core/pull/1700/commits/7b8a976a12c628da2d6bfa8cd3b2f353475d6e11.